src/main/java/org/jabref/logic/shared/listener/PostgresSQLNotificationListener.java

```diff
@@ -1,27 +1,50 @@
 package org.jabref.logic.shared.listener;
 
+import java.sql.SQLException;
+
 import org.jabref.logic.shared.DBMSProcessor;
 import org.jabref.logic.shared.DBMSSynchronizer;
 
-import com.impossibl.postgres.api.jdbc.PGNotificationListener;
+import org.postgresql.PGConnection;
+import org.postgresql.PGNotification;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * A listener for PostgreSQL database notifications.
  */
-public class PostgresSQLNotificationListener implements PGNotificationListener {
+public class PostgresSQLNotificationListener extends Thread {
 
-    private final DBMSSynchronizer dbmsSynchronizer;
+    private static final Logger LOGGER = LoggerFactory.getLogger(PostgresSQLNotificationListener.class);
 
+    private final DBMSSynchronizer dbmsSynchronizer;
+    private final PGConnection pgConnection;
 
-    public PostgresSQLNotificationListener(DBMSSynchronizer dbmsSynchronizer) {
+    public PostgresSQLNotificationListener(DBMSSynchronizer dbmsSynchronizer, PGConnection pgConnection) {
         this.dbmsSynchronizer = dbmsSynchronizer;
+        this.pgConnection = pgConnection;
     }
 
     @Override
-    public void notification(int processId, String channel, String payload) {
-        if (!payload.equals(DBMSProcessor.PROCESSOR_ID)) {
-            dbmsSynchronizer.pullChanges();
+    public void run() {
+        try {
+            //noinspection InfiniteLoopStatement
+            while (true) {
+                PGNotification notifications[] = pgConnection.getNotifications();
+
+                if (notifications != null) {
+                    for (PGNotification notification : notifications) {
+                        if (!notification.getName().equals(DBMSProcessor.PROCESSOR_ID)) {
+                            dbmsSynchronizer.pullChanges();
+                        }
+                    }
+                }
+
+                // Wait a while before checking again for new notifications
+                Thread.sleep(500);
```

Comment:
Doesn't the driver support push notifications?

Reply:
Frm the official doc it supports it:
https://jdbc.postgresql.org/documentation/head/listennotify.html#listen-notify-example